custom_components/grocy/__init__.py

```diff
@@ -4,151 +4,104 @@
 For more details about this integration, please refer to
 https://github.com/custom-components/grocy
 """
+from __future__ import annotations
+
 import logging
-from datetime import timedelta
 from typing import Any, List
 
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.core import Config, HomeAssistant
-from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
-from pygrocy import Grocy
-
-from .helpers import extract_base_url_and_path
+from homeassistant.core import HomeAssistant
 
 from .const import (
-    CONF_API_KEY,
-    CONF_PORT,
-    CONF_URL,
-    CONF_VERIFY_SSL,
+    ATTR_CHORES,
+    ATTR_EXPIRED_PRODUCTS,
+    ATTR_EXPIRING_PRODUCTS,
+    ATTR_MEAL_PLAN,
+    ATTR_MISSING_PRODUCTS,
+    ATTR_OVERDUE_CHORES,
+    ATTR_OVERDUE_TASKS,
+    ATTR_SHOPPING_LIST,
+    ATTR_STOCK,
+    ATTR_TASKS,
     DOMAIN,
-    GrocyEntityType,
     PLATFORMS,
     STARTUP_MESSAGE,
 )
-from .grocy_data import GrocyData, async_setup_image_api
+from .coordinator import GrocyDataUpdateCoordinator
+from .grocy_data import GrocyData, async_setup_endpoint_for_image_proxy
 from .services import async_setup_services, async_unload_services
 
-SCAN_INTERVAL = timedelta(seconds=30)
-
 _LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry):
     """Set up this integration using UI."""
-    hass.data.setdefault(DOMAIN, {})
     _LOGGER.info(STARTUP_MESSAGE)
 
-    coordinator = GrocyDataUpdateCoordinator(
-        hass,
-        config_entry.data[CONF_URL],
-        config_entry.data[CONF_API_KEY],
-        config_entry.data[CONF_PORT],
-        config_entry.data[CONF_VERIFY_SSL],
+    coordinator: GrocyDataUpdateCoordinator = GrocyDataUpdateCoordinator(hass)
+    coordinator.available_entities = await _async_get_available_entities(
+        coordinator.grocy_data
     )
-
     await coordinator.async_config_entry_first_refresh()
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][config_entry.entry_id] = coordinator
```

Comment:
This nested dictionary is only needed if the integration can be configured multiple times, which is not currently supported by this integration.